src/backtrace/mod.rs

```diff
@@ -57,10 +48,18 @@ pub(crate) fn print(
         .iter()
         .any(|raw_frame| raw_frame.is_exception());
 
-    let print_backtrace = settings.backtrace
-        || unwind.outcome == Outcome::StackOverflow
-        || unwind.corrupted
-        || contains_exception;
+    let print_backtrace= match settings.backtrace {
+        BacktraceOptions::Never => false,
+        BacktraceOptions::Always => true,
+        BacktraceOptions::Auto => if settings.panic_present
+            || unwind.outcome == Outcome::StackOverflow
+            || unwind.corrupted
+            || contains_exception {
+                true
+            } else {
+                false
```

Comment:
this `if...else` can be left out because the `settings.panic_present || unwind.outcome == Outcome::StackOverflow || ...` statement itself evaluates to a `bool`, which is what we want to return. e.g.:

```rust
 BacktraceOptions::Auto => settings.panic_present
             || unwind.outcome == Outcome::StackOverflow
             || unwind.corrupted
             || contains_exception 
```
should work as well